libs/pipelines/api_pipeline.py

```diff
@@ -80,60 +102,118 @@ def run_projections(
+    return APIOutput(generated_key, generated_data)
 
 
 def generate_api(
     projection_result: APIPipelineProjectionResult, input_dir: str
-) -> APIGeneration:
+) -> List[APIOutput]:
     """
     pipethrough the rows of the projection
     if it's a county generate the key for counties:
-        /us/counties/{FIPS_CODE}.{INTERVENTION}.json 
-    if it's a state generate the key for states 
+        /us/counties/{FIPS_CODE}.{INTERVENTION}.json
+    if it's a state generate the key for states
         /us/states/{STATE_ABBREV}.{INTERVENTION}.json
     """
     results = []
     for index, row in projection_result.projection_df.iterrows():
         results.append(_generate_api_without_ts(projection_result, row, input_dir))
         results.append(_generate_api_with_ts(projection_result, row, input_dir))
-    return APIGeneration(results)
-
-
-def deploy_results(result: APIGeneration, output: str):
+    return results
+
+
+def build_states_summary(state_data: List[APIOutput], intervention) -> APIOutput:
+    state_summaries = [
+        output.data
+        for output in state_data
+        # Don't love using type here, but CovidActNowStateTimeseries inherits
+        # CovidActNowStateSummary so `isinstance` returns true and fails to
+        # filter out the timeseries records.
+        if type(output.data) == CovidActNowStateSummary
+    ]
+    state_api_data = CovidActNowStatesSummary(data=state_summaries)
+    key = f"states.{intervention.name}"
+    return APIOutput(key, state_api_data)
+
+
+def build_states_timeseries(state_data: List[APIOutput], intervention) -> APIOutput:
+    state_summaries = [
+        output.data
+        for output in state_data
+        if isinstance(output.data, CovidActNowStateTimeseries)
+    ]
+    state_api_data = CovidActNowStatesTimeseries(data=state_summaries)
+    key = f"states.{intervention.name}.timeseries"
+    return APIOutput(key, state_api_data)
+
+
+def build_counties_summary(counties_data: List[APIOutput], intervention) -> APIOutput:
+    county_summaries = [
+        output.data
+        for output in counties_data
+        # Don't love using type here, but CovidActNowCountyTimeseries inherits
+        # CovidActNowCountySummary so `isinstance` returns true and fails to
+        # filter out the timeseries records.
+        if type(output.data) == CovidActNowCountySummary
+    ]
+    county_api_data = CovidActNowCountiesSummary(data=county_summaries)
+    key = f"counties.{intervention.name}"
+    return APIOutput(key, county_api_data)
+
+
+def build_counties_timeseries(
+    counties_data: List[APIOutput], intervention
+) -> APIOutput:
+    county_summaries = [
+        output.data
+        for output in counties_data
+        if isinstance(output.data, CovidActNowCountyTimeseries)
+    ]
+    county_api_data = CovidActNowCountiesTimeseries(data=county_summaries)
+    key = f"counties.{intervention.name}.timeseries"
+    return APIOutput(key, county_api_data)
```

Comment:
these all seem relatively duplicated, I think there's probably something we can do with like passing in classes or something